File: 2019W44/PreppindataW44.py
import pandas as pd

xls = pd.ExcelFile("E:/PD Wk 44 Input.xlsx")
sales = pd.read_excel(xls,'Store Sales')
members = pd.read_excel(xls,'Team Member Days')

sales = sales.melt(id_vars='Date', value_name='Store Sales', var_name='Store')

final = pd.merge(sales,members,how='inner',on=['Store','Date'])
final['Estimated Sales From Staff Member'] = final['Store Sales']/final['Team Member'].groupby([final['Store'],final['Date']]).transform('count')
final = final.groupby(['Store','Team Member'], as_index=False).agg({'Estimated Sales From Staff Member':'mean'}).round(2)
final['Rank'] = final.groupby('Store')['Estimated Sales From Staff Member'].rank(ascending=False).astype(int)

File: 2019W44/preppindataw44.R
library(readxl)
library(dplyr)
library(tidyr)

xls = "E:/PD Wk 44 Input.xlsx"

final <- read_excel(path = xls, sheet = 'Store Sales') %>%
  gather(., key='Store', value='Store Sales', -'Date',na.rm=TRUE) %>%
  merge(.,read_excel(path = xls, sheet = 'Team Member Days'), on=c('Store','Date')) %>%
  group_by(`Store`,`Date`) %>%
  mutate('Daily Avg Sales' = `Store Sales`/n()) %>%
  group_by(`Store`,`Team Member`) %>%
  summarise('Estimated Sales From Staff Member' = format(mean(`Daily Avg Sales`),nsmall=2)) %>%
  group_by(`Store`) %>%
  mutate('Rank' = rank(desc(as.numeric(`Estimated Sales From Staff Member`))))
  
View(final)  